### ISSUE TITLE
the full image results

### ISSUE THREAD

**Issue by minushuang** [chaneyddtt/Coarse-to-fine-3D-Animal#4]
the outputs now are generated from the cropped and transformed images, which can not be attached to the original images directly.
what should do if I want to the results aligned on the original images?
replaced the cropped images with original ? it seems not a good idea.
could you give me some instructions please.
many thanks.

**Comment by chaneyddtt** [comment]
Hi @minushuang, I do not think that replace the cropped image with the original would be a good idea. It will makes the training harder because the scale and location will varies a lot if directly using the original image. May i know why you need to align with the original image?

**Comment by minushuang** [comment]
Hi @chaneyddtt , thanks for your prompt reply.
for video inference scene, and more reasonable visualization results, I want to recovery the full image results  from current 224x224 cropped results, with `synth_rgb_refine` on it. 

I have digged into the code, and found that the input image of the model is not a simple bbox region of dog, it may have different scales and transforms, which makes the recovery processes some complicated.

is there any way that I can use to reach my scene?

**Comment by minushuang** [comment]
I will upload some figures tomorrow for better illustration and understanding.

**Comment by minushuang** [comment]
as the figure show below, they are original image, current model cropped result and the result I want(ignore the different synth_silhouette on it, and you can imagine a normal result with correct `synth_rgb_refine` output on it) respectively from left to right.
<img width="727" alt="image" src="https://user-images.githubusercontent.com/5811852/181139681-4ef4facc-dee3-42da-bf84-a4edebcefd8a.png">

**Comment by minushuang** [comment]
solved by get the transformed coordinates

**Comment by minushuang** [comment]
i can resize the crop result and attach it to original image according to the coordinates returned from function `mutils.crop`. it's ok but not the best I want in my scenario. 
my goal is to generate special video effects on the dog with the output 3d mesh in a realtime, does this project can do this ?
![image](https://user-images.githubusercontent.com/5811852/181212977-88a06fb4-668b-4919-b5a5-8c29f6e8c62f.png)

**Comment by chaneyddtt** [comment]
Yeah, you can transform back to the original image space in this way, and currently our model have not achieved real time inference yet.
